src/st/nz0/30958.c

```diff
@@ -363,7 +366,37 @@ INCLUDE_ASM("asm/us/st/nz0/nonmatchings/30958", EntitySlogra);
 INCLUDE_ASM("asm/us/st/nz0/nonmatchings/30958", func_801B54A8);
 
 // projectile fired from slogra's spear
-INCLUDE_ASM("asm/us/st/nz0/nonmatchings/30958", EntitySlograSpearProjectile);
+void EntitySlograSpearProjectile(Entity* self) {
+    Entity* newEntity;
+    if (self->unk34 & 0x100) {
+        newEntity = AllocEntity(D_8007D858, &D_8007D858[MaxEntityCount]);
+        if (newEntity != NULL) {
+            CreateEntityFromEntity(2U, self, newEntity);
+            newEntity->subId = 1;
+        }
+        DestroyEntity(self);
+        return;
+    }
+    switch (self->step) {
+    case 0:
+        InitializeEntity(&D_80180D24);
+        if (self->facing == 0) {
+            self->accelerationX = -0x40000;
+        } else {
+            self->accelerationX = 0x40000;
+        }
+    case 1:
+        if (AnimateEntity(&D_80181160, self) == 0) {
+            func_801BD52C(2U);
+            return;
+        }
+        return;
+    case 2:
+        MoveEntity();
+        AnimateEntity(&D_80181170, self);
+        break;
+    }
+}
```

Comment:
```suggestion
INCLUDE_ASM("asm/us/st/nz0/nonmatchings/30958", EntitySlograSpearProjectile);

Reply:
I recommend you revert this function to avoid having merge conflicts as this function is already merged to the repo.
I hope this doesn't discourage you much, as we only happened to casually worked on the same function.